### ISSUE TITLE
When installed alongside GitHub Copilot, OCaml Platform leads to an infinite extension-host crash loop

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [ocamllabs/vscode-ocaml-platform#1203]
Type: <b>Bug</b>

Reproduction:

1. Create a new 'temporary profile'
2. Install OCaml Platform
3. Install "GitHub Copilot Chat" (`GitHub.copilot-chat`, `v0.4.1`)
4. Reload VScode

This sends my machine into an endless extension-host crash.

<details>
<summary>
Log:
</summary>

```
log.ts:421  INFO Automatically restarting the remote extension host.
log.ts:421  INFO Creating a socket (renderer-ExtensionHost-8f3d253f-6922-4bc0-8d12-27fdfbc5c086)...
log.ts:421  INFO Creating a socket (renderer-ExtensionHost-8f3d253f-6922-4bc0-8d12-27fdfbc5c086) was successful after 166 ms.
log.ts:441   ERR No registered selector for ID: ms-vscode.npm-command: Error: No registered selector for ID: ms-vscode.npm-command
    at vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1515:26878
log.ts:421  INFO [remote-connection][ExtensionHost][8f3d2…][reconnect] received socket close event (wasClean: false, code: 1006, reason: ).
log.ts:441   ERR CloseEvent {isTrusted: true, wasClean: false, code: 1006, reason: '', type: 'close', …}
log.ts:421  INFO [remote-connection][ExtensionHost][8f3d2…][reconnect] starting reconnecting loop. You can get more information with the trace log level.
log.ts:421  INFO [remote-connection][ExtensionHost][8f3d2…][reconnect] resolving connection...
log.ts:421  INFO [remote-connection][ExtensionHost][8f3d2…][reconnect] connecting to WebSocket(127.0.0.1:60097)...
log.ts:421  INFO Creating a socket (renderer-ExtensionHost-8f3d253f-6922-4bc0-8d12-27fdfbc5c086)...
log.ts:421  INFO Creating a socket (renderer-ExtensionHost-8f3d253f-6922-4bc0-8d12-27fdfbc5c086) was successful after 176 ms.
log.ts:441   ERR [remote-connection][ExtensionHost][8f3d2…][reconnect][WebSocket(127.0.0.1:60097)] received error control message when negotiating connection. Error:
log.ts:441   ERR Error: Connection error: Unknown reconnection token (seen before)
    at q (remoteAgentConnection.ts:785:17)
    at s.value (remoteAgentConnection.ts:328:17)
    at d.w (event.ts:1088:13)
    at d.fire (event.ts:1119:9)
    at c.fire (ipc.net.ts:650:19)
    at b.z (ipc.net.ts:1021:28)
    at s.value (ipc.net.ts:955:72)
    at d.w (event.ts:1088:13)
    at d.fire (event.ts:1119:9)
    at s.acceptChunk (ipc.net.ts:394:21)
    at ipc.net.ts:350:51
    at s.value (browserSocketFactory.ts:230:39)
    at d.w (event.ts:1088:13)
    at d.fire (event.ts:1119:9)
    at i.onload (browserSocketFactory.ts:91:17)
log.ts:441   ERR [remote-connection][ExtensionHost][8f3d2…][reconnect] A permanent error occurred in the reconnecting loop! Will give up now! Error:
log.ts:441   ERR Error: Connection error: Unknown reconnection token (seen before)
    at q (remoteAgentConnection.ts:785:17)
    at s.value (remoteAgentConnection.ts:328:17)
    at d.w (event.ts:1088:13)
    at d.fire (event.ts:1119:9)
    at c.fire (ipc.net.ts:650:19)
    at b.z (ipc.net.ts:1021:28)
    at s.value (ipc.net.ts:955:72)
    at d.w (event.ts:1088:13)
    at d.fire (event.ts:1119:9)
    at s.acceptChunk (ipc.net.ts:394:21)
    at ipc.net.ts:350:51
    at s.value (browserSocketFactory.ts:230:39)
    at d.w (event.ts:1088:13)
    at d.fire (event.ts:1119:9)
    at i.onload (browserSocketFactory.ts:91:17)
abstractExtensionService.ts:757 Extension host (Remote) terminated unexpectedly. Code: 0, Signal: 8f3d253f-6922-4bc0-8d12-27fdfbc5c086
vb @ abstractExtensionService.ts:757
vb @ nativeExtensionService.ts:164
ub @ abstractExtensionService.ts:749
(anonymous) @ abstractExtensionService.ts:727
w @ event.ts:1088
fire @ event.ts:1119
v @ remoteExtensionHost.ts:201
(anonymous) @ remoteExtensionHost.ts:135
w @ event.ts:1088
fire @ event.ts:1119
fire @ ipc.net.ts:650
acceptDisconnect @ ipc.net.ts:979
H @ remoteAgentConnection.ts:774
v @ remoteAgentConnection.ts:729
u @ remoteAgentConnection.ts:723
t @ remoteAgentConnection.ts:677
await in t (async)
s @ remoteAgentConnection.ts:628
(anonymous) @ remoteAgentConnection.ts:595
w @ event.ts:1088
x @ event.ts:1099
fire @ event.ts:1123
fire @ ipc.net.ts:650
(anonymous) @ ipc.net.ts:876
g @ browserSocketFactory.ts:236
w @ event.ts:1088
fire @ event.ts:1119
(anonymous) @ browserSocketFactory.ts:174
log.ts:441   ERR Extension host (Remote) terminated unexpectedly with code 7.
log.ts:441   ERR Extension host (Remote) terminated unexpectedly. The following extensions were running: vscode.emmet, vscode.json-language-features, vscode.typescript-language-features, ocamllabs.ocaml-platform, vscode.git-base, vscode.npm, vscode.git, vscode.github, vscode.debug-auto-launch, vscode.merge-conflict, GitHub.copilot, GitHub.copilot-chat, GitHub.copilot-labs
notificationsAlerts.ts:42 Remote Extension host terminated unexpectedly 3 times within the last 5 minutes.
```

</details>

Extension version: 1.13.0
VS Code version: Code 1.80.2 (2ccd690cbff1569e4a83d7c43d45101f817401dc, 2023-07-27T21:05:41.366Z)
OS version: Darwin x64 22.5.0
Modes:
Connection to 'SSH: nspawn' could not be established  Canceled

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) W-3235 CPU @ 3.30GHz (24 x 3300)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|5, 5, 5|
|Memory (System)|48.00GB (1.88GB free)|
|Process Argv|--crash-reporter-id bfd1ab88-5618-4a53-98eb-52d3f905bb3a|
|Screen Reader|no|
|VM|0%|

Connection to 'SSH: nspawn' could not be established  Canceled
</details>

**Comment by toku-sa-n** [comment]
I had the same problem and resolved it by updating Copilot to v1.104.341.
Also, this issue seems to be a duplicate of #1186.

**Comment by rgrinberg** [comment]
Please re-open if you can reproduce in 1.13.1
